filter/filter_test.go

```diff
@@ -212,6 +215,51 @@ func TestFilters(t *testing.T) {
 		// Smoke test for globs...
 		{"s = hell*", records[15], true},
 		{"s = ell*", records[15], false},
+
+		// Test integer conditions.  These are really testing 2 things:
+		// 1. that the full range of values are correctly parsed
+		// 2. since the r.h.s. of a zql filter expression is treated
+		//    as int64, test that coercion happens correctly.
+		{"b > -1", records[17], true},
+		{"b = 0", records[17], true},
+		{"b < 1", records[17], true},
+		{"b > 1", records[17], false},
+
+		{"i16 = -32768", records[17], true},
+		{"i16 < 0", records[17], true},
+		{"i16 > 0", records[17], false},
+
+		{"u16 > -1", records[17], true},
+		{"u16 = 0", records[17], true},
+		{"u16 < 1", records[17], true},
+		{"u16 > 1", records[17], false},
+
+		{"i32 = -2147483648", records[17], true},
+		{"i32 < 0", records[17], true},
+		{"i32 > 0", records[17], false},
+
+		{"u32 > -1", records[17], true},
+		{"u32 = 0", records[17], true},
+		{"u32 < 1", records[17], true},
+		{"u32 > 1", records[17], false},
+
+		{"i64 = -9223372036854775808", records[17], true},
+		{"i64 < 0", records[17], true},
+		{"i64 > 0", records[17], false},
+
+		{"u64 > -1", records[17], true},
+		{"u64 = 0", records[17], true},
+		{"u64 < 1", records[17], true},
+		{"u64 > 1", records[17], false},
+
+		{"b = 255", records[18], true},
+		{"i16 = 32767", records[18], true},
+		{"u16 = 65535", records[18], true},
+		{"i32 = 2147483647", records[18], true},
+		{"u32 = 4294967295", records[18], true},
+		{"i64 = 9223372036854775807", records[18], true},
+		// can't represent large unsigned 64 bit values in zql...
```

Comment:
For my own info, not a comment on the PR: what prevents zql from handling large uint64's?

Reply:
> what prevents zql from handling large uint64's?
We don't have any syntax to indicate that an integer literal should be unsigned.  This is described in an issue on Jira, we could either automatically turn values that are too big to fit into int64 (but that do fit into uint64) or we could add new syntax (e.g., C style "1234U" or something)